app/src/main/java/com/example/wrk/TrackerActivity.java

```diff
@@ -133,12 +140,52 @@ private void publishWorkout() {
             public void done(ParseException e) {
                 if (e != null) {
                     Toast.makeText(TrackerActivity.this, "Error while publishing to feed.", Toast.LENGTH_SHORT).show();
-                    return;
                 }
             }
         });
     }
 
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    private void updateUserStreak() {
+        Date lastWorkout = ParseUser.getCurrentUser().getDate("lastWorkout");
+        int streak = ParseUser.getCurrentUser().getInt("streak");       // current daily streak
+        int dayDifference = 0;
+        int yearDifference = 0;
+        int currentDayOfYear;
+        int recentWorkoutDayOfYear;
+
+        // for working out on last day of year and 1st day of new year
+        final int NEW_YEAR_DIFF = -364;     // 1 - 365 = -364
+
+        LocalDate today = LocalDate.now();
+        currentDayOfYear = today.getDayOfYear();
+
+        // calculate difference in days since last workout
+        if (lastWorkout != null) {
+            // this is to allow for more than 24 hrs since workout as long as date is 1 day apart
+            LocalDate recentWorkout = lastWorkout.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+            recentWorkoutDayOfYear = recentWorkout.getDayOfYear();
+
+            dayDifference = currentDayOfYear - recentWorkoutDayOfYear;      // will be negative if two different years
+            yearDifference = today.getYear() - recentWorkout.getYear();     // to calculate if years have gone by since last workout
+        }
+
+        // increment streak only if there is a 1 day difference since the last workout.
+        // 1-365 = -364. If leap year then it would be -365.
```

Comment:
in order to not bother with the complicated calculations for last day of year and first day of year and leap years, could you try storing the start of the day timestamp and checking difference between timestamps? I assume there might still be issue if user travels and changes time zone, but still seems less hacky.

To get start of day, could we do something like
`today.toLocalDate().atStartOfDay( ZoneId.systemDefault() ).toEpochSecond() ;`

Reply:
Just changed it using the EpochSeconds. So far it looks like it works and is also isn't as complicated calculations.